graphql/schemabuilder/helpers.go

```diff
@@ -390,3 +405,111 @@ func castSlice(slice interface{}) []interface{} {
 
 	return ret
 }
+
+func (sb *schemaBuilder) buildPaginatedArgParser(originalArgType reflect.Type) (*argParser, graphql.Type, error) {
+
+	interfaceParser, interfaceType, err := sb.makeStructParser(originalArgType)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to build args for paginated field")
+	}
+
+	typ := reflect.TypeOf(ConnectionArgs{})
+	fields := make(map[string]argField)
+	argType := &graphql.InputObject{
+		Name:        typ.Name(),
+		InputFields: make(map[string]graphql.Type),
+	}
+	if argType.Name != "" {
+		argType.Name += "_InputObject"
+	}
+
+	if typ.Kind() != reflect.Struct {
+		return nil, nil, fmt.Errorf("expected struct but received type %s", typ.Name())
+	}
+	for i := 0; i < typ.NumField()-1; i++ {
+		field := typ.Field(i)
+		if field.PkgPath != "" && !field.Anonymous {
+			continue
+		}
+		if field.Anonymous {
+			return nil, nil, fmt.Errorf("bad arg type %s: anonymous fields not supported", typ)
+		}
+		tags := strings.Split(field.Tag.Get("graphql"), ",")
+		var name string
+		if len(tags) > 0 {
+			name = tags[0]
+		}
+		if name == "" {
+			name = makeGraphql(field.Name)
+		}
+		if name == "-" {
+			continue
+		}
+
+		var key bool
+
+		if len(tags) > 1 {
+			for _, tag := range tags[1:] {
+				if tag != "key" || key {
+					return nil, nil, fmt.Errorf("bad type %s: field %s has unexpected tag %s", typ, name, tag)
+				}
+				key = true
+			}
+		}
+
+		if _, ok := fields[name]; ok {
+			return nil, nil, fmt.Errorf("bad arg type %s: duplicate field %s", typ, name)
+		}
+		var parser *argParser
+		var fieldArgTyp graphql.Type
+		var err error
+		if field.Type.Kind() == reflect.Interface {
+			parser = interfaceParser
+			fieldArgTyp = interfaceType.(*graphql.InputObject)
+
+		} else {
+			parser, fieldArgTyp, err = sb.makeArgParser(field.Type)
+			if err != nil {
+				return nil, nil, err
+			}
+		}
+
+		fields[name] = argField{
+			field:  field,
+			parser: parser,
+		}
+		argType.InputFields[name] = fieldArgTyp
+	}
+
+	return &argParser{
+		FromJSON: func(value interface{}, dest reflect.Value) error {
+			asMap, ok := value.(map[string]interface{})
+			if !ok {
+				return errors.New("not an object")
+			}
+			interfaceFields := make(map[string]interface{})
```

Comment:
what is an interface field? it'd be helpful to comment on what is going into the pagination helper vs. going into the underlying resolver

Reply:
yeah going to rename a couple of variables + comment this part